tests/core/test_datamodules.py

```diff
@@ -533,6 +534,68 @@ def test_simple_hyperparameters_saving():
     assert data.hparams == AttributeDict({"arg0": 10, "arg1": "foo", "kwarg0": "bar"})
 
 
+def test_define_as_dataclass():
+    # makes sure that no functionality is broken and the user can still manually make
+    # super().__init__ call with parameters
+    # also tests all the dataclass features that can be enabled without breaking anything
+    @dataclass(init=True, repr=True, eq=True, order=True, unsafe_hash=True, frozen=False)
+    class BoringDataModule1(LightningDataModule):
+        batch_size: int
+        dims: int = 2
+
+        def __post_init__(self):
+            super().__init__(dims=self.dims)
```

Comment:
```suggestion
            super().__init__(dims=3)
```
don't we want to test this and assert below that dims = 3?

Reply:
Should it matter? As long as we are asserting for the value (2 or 3) that we are overwriting through `super().__init__` call, it should be fine right?